metricbeat/module/kafka/partition/partition.go

```diff
@@ -69,7 +71,8 @@ func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
 		BaseMetricSet: base,
 		broker:        broker,
 		cfg:           cfg,
-		id:            noID,
+		id:            config.ID,
```

Comment:
I'm actually considering to remove `id` and `advertised` settings. As these settings require `hosts` to be exactly one host.

Right, you can lookup the id. But I think it's much more convenient to not have to configure `id` or `advertised`. Problem with these options is, if they are wrongly configured (by accident), the kafka module will not work correctly.